src/world/area_kkj/kkj_19/AF7C00.c

```diff
@@ -4,29 +4,273 @@
+            D_8024655C += 20;
+            if (D_8024655C >= 51) {
+                D_8024655C = 50;
+            }
+            hud_element_set_render_pos(D_80246568_C8C018[0], D_8024655C, D_80246560_kkj_19);
+            hud_element_set_render_pos(D_80246568_C8C018[1], D_8024655C, D_80246560_kkj_19 + 28);
+            if (script->functionTemp[0] != 0) {
+                script->functionTemp[0]--;
+            } else {
+                D_80246558_kkj_19 = 10;
+            }
+            break;
+        case 10:
+            hud_element_set_script(D_80246568_C8C018[0], &HES_MashAButton);
+            dgb_08_npcGroup_80246528 = 0;
+            D_8024652C = 0;
+            D_80246558_kkj_19 = 11;
+            // fallthrough
+        case 11:
+            temp = D_80246538_kkj_19[D_80246550_kkj_19];
+            if (gGameStatusPtr->pressedButtons[0] & A_BUTTON) {
+                dgb_08_npcGroup_80246528++;
+            }
+            if (temp < dgb_08_npcGroup_80246528) {
+                dgb_08_npcGroup_80246528 = temp;
+            }
+            if (D_8024652C == 2) {
+                D_8024652C = 3;
+            }
+            for (i = 0; i < 10; i++) {
+                if (D_80246578[i] != 0) {
+                    break;
+                }
+            }
+            if (i >= 10) {
+                D_8024652C = 2;
+            } else if (D_8024652C != 3) {
+                D_8024652C = 2;
+            }
+            D_80246534--;
+            if (D_80246534 == 0) {
+                D_80246534 = 5;
+                D_80246558_kkj_19 = 12;
+            }
+            break;
+        case 12:
+            if (D_80246534 != 0) {
+                D_80246534--;
+                break;
+            }
+            script->varTable[0] = dgb_08_npcGroup_80246528;
+            hud_element_free(D_80246568_C8C018[0]);
+            hud_element_free(D_80246568_C8C018[1]);
+            free_generic_entity(D_80246554);
+            return ApiStatus_DONE2;
+    }
+
+    D_80246578[D_802465A0++] = gGameStatusPtr->pressedButtons[0] & A_BUTTON;
+
+    if (D_802465A0 >= 10) {
+        D_802465A0 = 0;
+    }
+
+    evt_set_variable(script, -0x08F0D17D, D_8024652C);
+    return ApiStatus_BLOCK;
+}
 
 static char* N(exit_str_0) = "kkj_18";
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_19/AF7C00", func_80240B4C_AF849C);
+ApiStatus func_80240B4C_AF849C(Evt* script, s32 isInitialCall) {
+    if (isInitialCall) {
+        script->functionTemp[1] = 0;
+    }
+
+    if (gGameStatusPtr->pressedButtons[0] & BUTTON_A) {
+        script->varTable[0] = script->functionTemp[1];
+        return ApiStatus_DONE2;
+    }
+
+    script->functionTemp[1]++;
+    return ApiStatus_BLOCK;
+}
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_19/AF7C00", func_80240B8C_AF84DC);
+ApiStatus func_80240B8C_AF84DC(Evt* script, s32 isInitialCall) {
+    Bytecode args = *script->ptrReadPos;
+    s32 itemID = evt_get_variable(script, args);
+
+    evt_set_variable(script, args, gItemTable[itemID].nameMsg);
```

Comment:
```suggestion
    evt_set_variable(script, *args++, gItemTable[itemID].nameMsg);
```

Reply:
For this one in particular I had to use the same arg in `evt_get_variable` and in `evt_set_variable`, so I changed it to:
```c
    Bytecode* args = script->ptrReadPos;
    Bytecode var = *args++;
    s32 itemID = evt_get_variable(script, var);

    evt_set_variable(script, var, gItemTable[itemID].nameMsg);
    return ApiStatus_DONE2;
```